File: Maternal_Mortality_Project/scripts/descriptive_analysis.py
import pandas as pd, seaborn as sns, matplotlib.pyplot as plt

df = pd.read_csv('../data/maternal_mortality_clean.csv')
race_summary = df.groupby("Race/ethnicity")["MMR"].mean().reset_index()
borough_summary = df.groupby("Borough")["MMR"].mean().reset_index()

sns.lineplot(data=df, x="Year", y="MMR", hue="Race/ethnicity", marker="o")
plt.title("MMR by Race/Ethnicity"); plt.savefig('../results/MMR_by_Race_Py.png'); plt.clf()

sns.barplot(data=borough_summary, x="Borough", y="MMR")
plt.title("Average MMR by Borough"); plt.savefig('../results/MMR_by_Borough_Py.png')

File: Maternal_Mortality_Project/scripts/descriptive_analysis.R


library(tidyverse)
library(ggplot2)

# Load cleaned dataset
df <- read_csv("maternal_mortality_clean.csv")

# Summary by race
race_summary <- df %>%
  group_by(`Race/ethnicity`) %>%
  summarise(Total_Deaths = sum(Deaths, na.rm=TRUE))

# Summary by borough
borough_summary <- df %>%
  group_by(Borough) %>%
  summarise(Total_Deaths = sum(Deaths, na.rm=TRUE))

# Save summary tables
write_csv(race_summary, "Race_Disparities_R.csv")
write_csv(borough_summary, "Borough_Disparities_R.csv")

# Plot: Deaths by Race over time
p1 <- ggplot(df, aes(x=Year, y=Deaths, color=`Race/ethnicity`)) +
  geom_line() + geom_point() +
  labs(title="Maternal Deaths by Race/Ethnicity (NYC, 2010–2024)",
       y="Number of Deaths") +
  theme_minimal()
ggsave("Deaths_by_Race_R.png", plot=p1, width=8, height=5, dpi=300)

# Plot: Deaths by Borough
p2 <- ggplot(borough_summary, aes(x=Borough, y=Total_Deaths)) +
  geom_col(fill="steelblue") +
  labs(title="Maternal Deaths by Borough", y="Total Deaths") +
  theme_minimal()
ggsave("Deaths_by_Borough_R.png", plot=p2, width=8, height=5, dpi=300)
